tests/utils/autoTester.js

```diff
@@ -184,65 +235,26 @@ var doAddStep = function (name, stepFunction, snapshot) {
 //
 // The method creates a test case in currentTestCase.
 // If the test mode is manual or auto then its execution is started
-module.exports.perform = function (testName, testPath, testMain) {
-    currentTestCase = new TestCase(testName, testPath, testMain);
-    
-    // Manual and auto tests are run immediately, recursive tests are stored in a queue
-    if (isRecursive) {
-        print("Not running yet - in recursive mode");
-        testCases.push(currentTestCase);
-    } else if (testMode === "manual") {
-        print("Begin manual test:" + testName);
-        currentTestCase.func("manual");
-    } else { // testMode === "auto"
-        print("Begin auto test:" + testName);
-        currentTestCase.func("auto");
-    }
-}
+module.exports.perform = function (testName, testPath, validationCamera, testMain) {
+    var usePrimaryCamera = (validationCamera === "primary");
 
-module.exports.setupTest = function (usePrimaryCameraForSnapshots) {
-    if (currentTestCase === null) {
-        return;
-    }
+    currentTestCase = new TestCase(testName, testPath, testMain, usePrimaryCamera);
```

Comment:
As we discussed yesterday, this code that was in setupTest needs to be executed Just before when the FUnctor of the test case run
otherwise, this won;t be equivalent to what was happening in the previous version of the code
You could simply keep the setup function 
And call it JUST before calling the TestCase Function